### DISCUSSION TITLE
x-data @entangle(events) json

### DISCUSSION THREAD

**Discussion by jouniikaheimo** [alpinejs/alpine#1023] [1. Help]
livewire public $events is json .... my code is 
            `<div x-data='{...app(), events: @entangle(events) }' x-init="[initDate(), getNoOfDays()]" x-cloak>`
i got error: syntax error, unexpected '->' (T_OBJECT_OPERATOR), expecting ')' 

if i change that:             
`<div x-data='{...app(), events: @json($events) }' x-init="[initDate(), getNoOfDays()]" x-cloak>`
and remove livewire json_encode that working fine.. but if i add new events and emit listener protected 
```
$listeners = [
        'calendar:update' => '$refresh',
    ];
```
calendar events not refresh how i can refresh x-data json after new events save ?

**Comment by KevinBatdorf** [comment]
Try updating `@entangle(events)` to `@entangle('events')`. Any difference?
